aten/src/ATen/core/RRef.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include <c10/util/intrusive_ptr.h>
+
+namespace c10 {
+
+struct Type;
+using TypePtr = std::shared_ptr<Type>;
+using worker_id_t = int16_t;
+
+// This abstract class contains only user-facing APIs, and will be shared
+// between jit and distributed to implement TorchScript support.
+class C10_EXPORT RRef : public c10::intrusive_ptr_target {
```

Comment:
Shouldn't this class be called `RRefInterface` and we can call `RRefBase` simply `RRef`? Also, should the comment block in RRef.h explaining RRefs move here?

Reply:
yeah I was just pulling the changes from https://github.com/pytorch/pytorch/pull/28942 to test it initially. `RRefInterface` is probably a better name I agree.